src/index.js

```diff
@@ -44,65 +42,72 @@ exports.install = function (Vue, options = {}) {
         default () {
           return []
         }
-      }
+      },
+      noDataOptions: {
+        type: Object,
+        default () {
+          return {
+            message: '',
+            class: 'ct-nodata'
+          }
+        }
+      },
     },
     data () {
       return {
         chart: null,
-        error: { onError: false, message: '' },
         noData: false,
         message: '',
-        classNoData: options.classNoData
       }
     },
     watch: {
       ratio: 'redraw',
       options: { handler: 'redraw', deep: true },
+      responsiveOptions: { handler: 'redraw', deep: true },
       data: { handler: 'redraw', deep: true },
       type: 'draw',
-      eventHandlers: 'resetEventHandlers'
+      eventHandlers: 'resetEventHandlers',
+      haveNoData: {
+        immediate: true,
+        handler: function (val) {
+          if (val) {
+            this.setNoData()
+          } else {
+            this.clear()
+          }
+        }
+      }
     },
     mounted () {
       this.draw()
     },
+    computed: {
+      haveNoData: function () {
+        return !this.data ||
+            !this.data.series ||
+            this.data.series.length < 1 ||
+            (
+                (this.type !== 'Pie' && !this.options.distributeSeries) &&
+                this.data.series.every(series => {
+                  if (Array.isArray(series)) {
+                    return !series.length
+                  }
+                  return !series.data.length
+                })
+            )
+      },
+    },
```

Comment:
We can then add a `noDataOptions` computed property

```
return {
        message: options.message || this.noDataProp.message,
        class: options.class || this.noDataProp.class
  }
```

This way you give priority to global options if they're set